src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        int numberOfFigures = 5;
+        int randomFigure = random.nextInt(numberOfFigures);
+
+        switch (randomFigure) {
+            case 0:
+                figure = new Circle(colorSupplier.getRandomColor(), getRandomSide());
+                break;
+            case 1:
+                figure = new IsoscelesTrapezoid(colorSupplier.getRandomColor(), getRandomSide(),
+                        getRandomSide(), getRandomSide());
+                break;
+            case 2:
+                figure = new Rectangle(colorSupplier.getRandomColor(), getRandomSide(),
+                        getRandomSide(), getRandomSide());
+                break;
+            case 3:
+                figure = new RightTriangle(colorSupplier.getRandomColor(), getRandomSide(),
+                        getRandomSide());
+                break;
+            default:
+                figure = new Square(colorSupplier.getRandomColor(), getRandomSide());
+                break;
+        }
+
+        return figure;
+    }
+
+    private double getRandomSide() {
+        int maxSideUnits = 9;
+        return maxSideUnits * random.nextDouble() + 1;
+    }
+
+    public Figure getDefaultFigure() {
```

Comment:
> Are you sure that this method is needed, some default figure returning logic defined in 'getRandomFigure' method

Didn't understand your message, because, in the README file, we have a task about this method exist,